source/features/limit-commit-title-length.tsx

```diff
@@ -0,0 +1,50 @@
+import './limit-commit-title-length.css';
+import select from 'select-dom';
+import features from '../libs/features';
+import onPrMergePanelOpen from '../libs/on-pr-merge-panel-open';
+
+function init(): void {
+	const inputs = select.all<HTMLInputElement>([
+		'#commit-summary-input', // Commit title on edit file page
+		'#merge_title_field' // PR merge message field
+	].join(','));
+
+	for (const inputField of inputs) {
+		const validityCallback = (event: Event): void => {
+			if (inputField.value.length > 72) {
+				event.preventDefault();
+				event.stopImmediatePropagation();
+
+				inputField.setCustomValidity(`The title should be maximum 72 characters, but is ${inputField.value.length}`);
+
+				if (event.type === 'submit') {
+					inputField.reportValidity();
+				}
+			} else {
+				inputField.setCustomValidity('');
+			}
+		};
+
+		inputField.addEventListener('input', validityCallback);
+		inputField.form!.addEventListener('submit', validityCallback);
+	}
+
+	// For PR merges, GitHub restores any saved commit messages on page load
+	// Triggering input event for these fields immediately validates the form
+	onPrMergePanelOpen(() => {
+		for (const inputField of inputs) {
+			inputField.dispatchEvent(new Event('input'));
+		}
+	});
```

Comment:
Good idea, I wonder if this makes the `submit` handler unnecessary. I don't think the title field is ever programmatically changed when the PR merge panel is open; even GitHub specifically avoids this. 

In that case, validityCallback can be simplified as I had first suggested, likely.

Also this event only needs to be triggered on the title field because that's what this feature cares about.

Reply:
> In that case, validityCallback can be simplified as I had first suggested, likely.

Yup!

-------------

> Also this event only needs to be triggered on the title field because that's what this feature cares about.

The merge input field can be on index 0 or 1 of the `inputFields` variable when querying DOM. Instead of searching for it, I just figured it'd be easy to trigger input events on all of them. Doesn't matter that much anyway.

Reply:
I had assumed that `fields` was `[title, message]` but I think it's always just one; the double selector made me thing it was two. I think you just need a `select` and you can drop the loop